Common/ExecutionModel/vtkThreadedImageAlgorithm.cxx

```diff
@@ -25,12 +25,105 @@
 #include "vtkPointData.h"
 #include "vtkStreamingDemandDrivenPipeline.h"
 
+#include "vtkExtentTranslator.h"
+
+
+
+//---- option 2
+
+     #include <tbb/parallel_for.h>
+#include <tbb/blocked_range.h>
+#include <tbb/blocked_range3d.h>
+#include <tbb/blocked_range2d.h>
+#include <tbb/task_scheduler_init.h>
+//-- option 1
+#include "vtkSMPTools.h"
+
+
+
+
+
+
+struct vtkImageThreadStruct
+{
+vtkThreadedImageAlgorithm *Filter;
+vtkInformation *Request;
+vtkInformationVector **InputsInfo;
+vtkInformationVector *OutputsInfo;
+vtkImageData   ***Inputs;
+vtkImageData   **Outputs;
+};
+
+class apply_transform{
+  
+public:
+  vtkImageThreadStruct* str;
+  apply_transform(vtkImageThreadStruct* abc)
+  {
+    str = abc;
+  }
+  apply_transform()
+  {
+   
```

Comment:
All constructors should initialize all member variables.  Unlike Java, variables are not automatically assigned a "default value".  Instead, C++ leaves them uninitialized.